src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PublisherEventsPersistorTest.java

```diff
@@ -47,23 +47,68 @@ public void cleanup() throws IOException {
     }
 
     @Test
-    public void actionPublished() throws InvocationTargetException, NoSuchMethodException, InstantiationException, IllegalAccessException {
-        final MockAction mockAction = new MockAction();
-
-        publisherEventsPersistor.persistAction(mockAction);
-
-        PersistedAction actionsSummary = persistable.read(PersistedAction.class);
+    public void actionPublished() throws Exception {
+        final MockAction mockAction1 = new MockAction("MockAction1", new ArrayList<String>() {
+            {
+                add("1");
+                add("11");
+            }});
+        final MockAction mockAction2 = new MockAction("MockAction2", new ArrayList<String>() {
+            {
+                add("2");
+                add("33");
+            }});
+        List<Action> mockActions = new ArrayList<>();
+        mockActions.add(mockAction1);
+        mockActions.add(mockAction2);
+        mockActions.clear();
+
+        publisherEventsPersistor.persistAction(mockActions);
+
+        final MockAction mockAction3 = new MockAction("MockAction3",new ArrayList<String>() {
+            {
+                add("3");
+                add("33");
+            }});
+        final MockAction mockAction4 = new MockAction("MockAction4",new ArrayList<String>() {
+            {
+                add("4");
+                add("44");
+            }});
+        mockActions.add(mockAction3);
+        mockActions.add(mockAction4);
+
+        publisherEventsPersistor.persistAction(mockActions);
+
+        WaitFor.waitFor(() -> persistable.readLatestGroup(PersistedAction.class).size() == 2, 5,
+                TimeUnit.SECONDS);
+        List<PersistedAction> actionsSummary = persistable.readLatestGroup(PersistedAction.class);
         Assert.assertNotNull(actionsSummary);
-        Assert.assertEquals(actionsSummary.getActionName(), mockAction.name());
-        Assert.assertEquals(actionsSummary.getNodeIds(), "{1,2}");
-        Assert.assertEquals(actionsSummary.getNodeIps(), "{1.1.1.1,2.2.2.2}");
-        Assert.assertEquals(actionsSummary.isActionable(), mockAction.isActionable());
-        Assert.assertEquals(actionsSummary.getCoolOffPeriod(), mockAction.coolOffPeriodInMillis());
-        Assert.assertEquals(actionsSummary.isMuted(), mockAction.isMuted());
-        Assert.assertEquals(actionsSummary.getSummary(), mockAction.summary());
+        Assert.assertEquals(actionsSummary.size(), 2);
```

Comment:
Can we check that the table itself had 4 action entries at this point with 2 different sets of timestamps?

Reply:
Thats right, Vigya, the bug in the UT was I was clearing the actionSummary before it was writing in the table. Updated the UT.